standalone/pherry/src/main.rs

```diff
@@ -674,6 +675,71 @@ async fn init_runtime(
     Ok(resp)
 }
 
+async fn get_geolocation() -> Result<CoordinateInfo> {
+    let res = reqwest::get("https://ipinfo.io/ip").await?;
+    if !res.status().is_success() {
+        return Err(anyhow::Error::msg(
+            format!("Cannot get public IP: {}", res.status())
+        ));
+    }
+    let pub_ip = res.text().await?;
+    info!("Public IP address: {}", pub_ip);
+
+    use maxminddb::geoip2;
+    use std::net::IpAddr;
+
+    let geo_db_bytes = include_bytes!("../GeoLite2-City.mmdb");
+    let geo_db_buf: Vec<u8> = geo_db_bytes.to_vec();
+    let reader = maxminddb::Reader::from_source(geo_db_buf).unwrap();
```

Comment:
Let's minimize the usage of `unwrap()`. If we feel confident it will never fall, use `expect("... the reason; qed.")`

Reply:
Done